ipapython/ipa_log_manager.py

```diff
@@ -141,12 +160,45 @@ def filter(self, record):
                 record.levelno >= self.level)
 
 
+class InfoCommandFilter(logging.Filter):
+    """
+    The InfoCommandFilter filters out info_cmd log messages for the log file.
+    info_cmd log messages should only be visble on the command line.
+    """
+
+    def filter(self, record):
+        if hasattr(record, "cmd_output") and record.__dict__["cmd_output"]:
+            return False
+        return True
+
+
 class Formatter(logging.Formatter):
     def __init__(
             self, fmt=LOGGING_FORMAT_STDOUT, datefmt=ISO8601_UTC_DATETIME_FMT):
         super(Formatter, self).__init__(fmt, datefmt)
         self.converter = time.gmtime
 
+    def format(self, record):
+        """
+        Format the specified record as text.
+
+        Uses LOGGING_FORMAT_INFO_CMD for info_cmd, otherwise normal logging
+        format. For info_cmd record.__dict__["cmd_output"] is set to True.
+        """
+        if record.levelno == logging.INFO and \
+           hasattr(record, "cmd_output") and record.__dict__["cmd_output"]:
+            # Save original logging format
+            orig_fmt = self._fmt
+            # Temporarily use LOGGING_FORMAT_INFO_CMD
+            self._fmt = LOGGING_FORMAT_INFO_CMD
+            # Create result
+            result = logging.Formatter.format(self, record)
```

Comment:
```
try:
    return super(Formatter, self).format(record)
finally:
    self._fmt = orig_fmt
```

Reply:
Solved in a different way with an special internal Formatter for cmd_output.